gitops/environment.go

```diff
@@ -0,0 +1,100 @@
+package gitops
+
+import (
+	"github.com/google/uuid"
+	applicationapiv1alpha1 "github.com/redhat-appstudio/application-api/api/v1alpha1"
+	"github.com/redhat-appstudio/integration-service/api/v1alpha1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type CopiedEnvironment struct {
+	applicationapiv1alpha1.Environment
+}
+
+// AsPipelineRun casts the IntegrationPipelineRun to PipelineRun, so it can be used in the Kubernetes client.
+func (r *CopiedEnvironment) AsEnvironment() *applicationapiv1alpha1.Environment {
+	return &r.Environment
+}
+
+// check if Env already contains requested Name
+func contains(s applicationapiv1alpha1.EnvironmentConfiguration, e string) bool {
+	for a := range s.Env {
+		if s.Env[a].Name == e {
+			return true
+		}
+	}
+	return false
+}
+
+// NewCopyOfExistingEnvironment gets the existing environment from current namespace and makes copy of it
+// new name is generated consisting of existing environment name and integrationTestScenario name
+// targetNamespace gets name of integrationTestScenario and uuid
+func NewCopyOfExistingEnvironment(existingEnvironment *applicationapiv1alpha1.Environment, namespace string, integrationTestScenario *v1alpha1.IntegrationTestScenario, snapshot *applicationapiv1alpha1.Snapshot) *CopiedEnvironment {
```

Comment:
The `snapshot` parameter is unused, I'm guessing it's not needed since there's `WithSnapshot` below.